### ISSUE TITLE
Shader compilation exceptions not holding messages

### ISSUE THREAD

**Issue by rgerd** [BabylonJS/BabylonNative#733]
**Describe the bug**
The following exceptions (for example):
https://github.com/BabylonJS/BabylonNative/blob/432f0db8e0b0ddd98fb9c7744572f87c9c16528b/Plugins/NativeEngine/Source/ShaderCompilerD3D.cpp#L23-L26
https://github.com/BabylonJS/BabylonNative/blob/432f0db8e0b0ddd98fb9c7744572f87c9c16528b/Plugins/NativeEngine/Source/ShaderCompilerD3D.cpp#L93-L96

Don't hold their error messages (the "what" field) when being rethrown here:
https://github.com/BabylonJS/BabylonNative/blob/432f0db8e0b0ddd98fb9c7744572f87c9c16528b/Plugins/NativeEngine/Source/NativeEngine.cpp#L637-L644

Seems like this is just a case of memory getting freed, so should (hopefully) be a pretty straightforward issue to fix.

**To Reproduce**
I triggered the linker exception by doing something like:
`shader.vertex.fx`
```glsl
uniform float _Radius_Top_Left_;
```
`shader.fragment.fx`
```glsl
mediump uniform float _Radius_Top_Left_;
```

And I triggered the parser exception by trying to compile a shader missing a paren.

**Expected behavior**
The exception being thrown should hold the error message and show up in the javascript console. Right now the error message is blank:
<img width="1028" alt="empty-error-bjs-shader" src="https://user-images.githubusercontent.com/4724014/117338897-f45a2880-ae53-11eb-8b53-01aea6797d78.PNG">

**Screenshots**
The "what" of the error here is "":
<img width="561" alt="shader-error-what-none" src="https://user-images.githubusercontent.com/4724014/117338556-93caeb80-ae53-11eb-8d3a-378475b2d1f5.PNG">
But the info sink holds useful info:
<img width="567" alt="shader-compiler-info-sink" src="https://user-images.githubusercontent.com/4724014/117338552-9299be80-ae53-11eb-89b5-f748020fd08e.PNG">
Same here with the parser:
<img width="750" alt="parse-error-info" src="https://user-images.githubusercontent.com/4724014/117339543-b578a280-ae54-11eb-872f-96bc9e3ef2e2.PNG">

**Other**
 - Platform: Win32

**Comment by syntheticmagus** [comment]
Under the specified circumstances, the empty "what" being being bubbled up is expected from the current implementation because `getDebugInfoLog()` returns the empty string; the actual compilation error message can be retrieved by `getInfoLog()`. I'm not sure if we know of a circumstance where `getDebugInfoLog()` will contain information we need -- if so, we should add it back -- but assuming we're just grabbing the wrong log, I swapped them out in #772, which should address this issue.

**Comment by syntheticmagus** [comment]
@rgerd, can you confirm that this is fixed by #772 and, if so, close this issue?
